src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/SessionPass.php

```diff
@@ -22,16 +22,31 @@ class SessionPass implements CompilerPassInterface
 {
     public function process(ContainerBuilder $container)
     {
-        if (!$container->hasDefinition('session')) {
+        if (!$container->has('.session.do-not-use')) {
             return;
         }
 
+        // BC layer: Make "session" an alias of ".session.do-not-use" when not override by user
+        if (!$container->has('session')) {
+            $alias = $container->setAlias('session', '.session.do-not-use');
+            $alias->setDeprecated('symfony/framework-bundle', '5.2', 'The "%alias_id%" alias is deprecated, use "$requestStack->getSession()" instead.');
+
+            return;
+        }
+
+        $definition = $container->getDefinition('session');
```

Comment:
`findDefinition`

Reply:
but we should either deprecate the definition or the alias, not the resolved definition

Reply:
right